### DISCUSSION TITLE
Some forks not appearing on metrics

### DISCUSSION THREAD

**Discussion by eduardosantoshf** [lowlighter/metrics#923] [Q&A]
Hey! :)

Yesterday, I started using metrics on my github profile README, and I noticed something strange: some of my forks are not appearing on the metrics. 

Here are my commits (including forks) of the previous days: 

<img width="85" alt="Screen Shot 2022-02-28 at 00 50 50" src="https://user-images.githubusercontent.com/45774025/155907965-b1d1fe01-e0ba-417c-84d4-00acd178c6cf.png">

And here is the metrics commits:

<img width="397" alt="Screen Shot 2022-02-28 at 00 51 01" src="https://user-images.githubusercontent.com/45774025/155907991-fba46ede-3ef8-4bff-9fe2-6422e7884c20.png">

The strangest thing is that, for example, in one day I only made one fork (0 commits) and it appeared on both my github contributions graph and on the metrics contributions, but in the next day I also only made one fork (0 commits) and that day appears as "empty" on the metrics contributions.

(It is relevant to mention that both forks were made from the same github user)

Am I missing some detail?

Thanks in advance for the help! :)

**Comment by lowlighter** [comment]
Hi 👋 !

When you're talking about forks, you mean repo you've forked ?
If yes, it may be related to the fact that forks are excluded by default, but you can include them with [`repositories_forks`](https://github.com/lowlighter/metrics/blob/master/source/plugins/base/README.md#repositories_forks) option. 

Additionally, fetched repos may differ slightly from what GitHub displays based on the chosen [`repositories_affiliations`](https://github.com/lowlighter/metrics/blob/master/source/plugins/base/README.md#repositories_affiliations), which supports contraints such as `owner`, `collaborator` or `organization_member`

Hope this helps 🙂

**Comment by eduardosantoshf** [comment]
Hello!

Thanks for answering! 🙂

Yes, I'm talking about repo forks. I already included the `repositories_forks` option, as well as the `repositories_affiliations` option, with all its constraints, and the problem persistst 😕

The strange thing is that some forked repos that appear on the metrics and some that don't appear are "similar" repos from the same GitHub user.

**Reply by lowlighter** [reply]
I'm not sure either, there aren't a lot of post-processing for the 2-weeks calendar, it uses directly the result from this query:
https://github.com/lowlighter/metrics/blob/dda33cfde8f47732fb6615541b2202bfc1835281/source/plugins/base/queries/user.x.graphql#L45-L53

And we apply `contributionCalendar.weeks.flatMap(({contributionDays}) => contributionDays).slice(-14)` on the result.

___

If you have some time, maybe you could try to check wether you obtain the same results as the commits graph from the [GitHub GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer) by using the query above (replacing values with your own)
